weatherman.py

```diff
@@ -0,0 +1,331 @@
+import os
+from collections import namedtuple
+from datetime import datetime
+import argparse
+
+
+# consts for indexes in WeatherReading tuple
+DATE_INDEX = 0
+MAX_TEMP_INDEX = 1
+MEAN_TEMP_INDEX = 2
+MIN_TEMP_INDEX = 3
+MAX_HUMIDITY_INDEX = 4
+MEAN_HUMIDITY_INDEX = 5
+MIN_HUMIDITY_INDEX = 6
+
+# data structure to hold weather readings for a given day
+WeatherReading = namedtuple('WeatherReading', [
+    'date', 'max_temp', 'mean_temp', 'min_temp', 'max_humidity', 'mean_humidity', 'min_humidity'
+])
+
+
+class DictDataItem:
+    ''' Dictionary data item to store key value pairs '''
+
+    def __init__(self):
+        self.data = {}
+
+    def add_data(self, key, value):
+        self.data[key] = value
+
+    def get_data(self, key):
+        return self.data[key]
+
+    def __str__(self):
+        return str(self.data)
+
+
+class WeatherDataParser:
+    ''' Parser class which populates the data rows '''
+
+    def __init__(self, folder_path):
+        self.weather_reading_list = []
```

Comment:
We don't need to add type of data structure at the end, `weather_readings` would have been good enough as well